### PR TITLE
Add sampling to logtracing

### PR DESCRIPTION
**Pull request by w11th**
These features have been added to `logtracing`:
1. Set up the default sampler
2. Set up a sampler when starting a new span
3. Log a sampled span with tag `is_sampled=1`

### CODE REVIEW

**Comment by w11th** [comment]
@bodhi Could you please have a review when you have time?

File: logtracing/sampling.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	return func(p SamplingParameters) bool {
+		return true
+	}
+}
+
+func NeverSample() Sampler {
+	return func(p SamplingParameters) bool {
+		return false
+	}
+}
```
Comment (bodhi) [inline logtracing/sampling.go:43]: This function only exists to be tested? It's never used.

File: logtracing/span.go [hunk 2]
```diff
@@ -74,6 +82,15 @@ func (s *span) AppendKVs(keyvals ...interface{}) {
 	s.keyvals = append(s.keyvals, keyvals...)
 }
 
+func (s *span) Meta() SpanMeta {
```
Comment (bodhi) [inline logtracing/span.go:85]: Does this need to be exported?

File: logtracing/span.go [hunk 3]
```diff
@@ -9,12 +9,20 @@ import (
 	"time"
 )
 
+type SpanMeta struct {
```
Comment (bodhi) [inline logtracing/span.go:12]: As for `func Meta()`, does this type need to be exported?

**Review by bodhi — APPROVED** [review]
Looks reasonable to me, my only question is about whether exporting metadata from the package is needed. Keeping the public API small is nice.

File: logtracing/span.go (hunk 2 above)
Comment (w11th) [reply logtracing/span.go]: Right, there are no other scenarios to use this now. I'll update this.

File: logtracing/span.go (hunk 3 above)
Comment (w11th) [reply logtracing/span.go]: Thanks, I'll update this.

File: logtracing/sampling.go (hunk 1 above)
Comment (w11th) [reply logtracing/sampling.go:43]: 🤔 It's a simple helper function that I thought it might be useful when I want to "mute" spans.

**Pull request merged**